### PR TITLE
Plots inspector using Vue

### LINKED ISSUE #3687: [Plots] Angular Plot parity
Issues to be addressed in Vue plots to achieve parity with Angular plots
* [x] Plot grids do not show in the new Vue plots
* [ ] The cursor should change from a cross to a hand when the pan modifier key is held down (same as conductor axis)
* [x] Remove `mct` prefix from all files, class, and component names
* [x] Inspector needs to be re-implemented sans Angular

### PR DESCRIPTION
**Pull request by shefalijoshi**
Remove use of Angular in Plots inspector views
Resolves #3687

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y (and adding more)
3. Command line build passes? Y
4. Changes have been smoke-tested? Y
5. Testing instructions included? Y

### CODE REVIEW

File: src/plugins/timeline/TimelineViewLayout.vue [hunk 1]
```diff
@@ -52,30 +52,18 @@
...
         </div>
     </div>
 </div>
 </template>
 
 <script>
-import ObjectView from '@/ui/components/ObjectView.vue';
+import TimelineObjectView from './TimelineObjectView.vue';
 import TimelineAxis from '../../ui/components/TimeSystemAxis.vue';
 import SwimLane from "@/ui/components/swim-lane/SwimLane.vue";
```
Comment (nikhilmandlik) [inline src/plugins/timeline/TimelineViewLayout.vue:68]: I think can remove this SwimLane import?

File: src/plugins/timeline/TimelineViewLayout.vue [hunk 2]
```diff
@@ -101,7 +89,7 @@ function getViewKey(domainObject, objectPath, openmct) {
 
 export default {
     components: {
-        ObjectView,
+        TimelineObjectView,
         TimelineAxis,
         SwimLane
```
Comment (nikhilmandlik) [inline src/plugins/timeline/TimelineViewLayout.vue:94]: same, remove this?

File: src/plugins/plot/vue/inspector/PlotOptions.vue [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ licenses. See the Open Source Licenses file (LICENSES.md) included with
+ this source code distribution or the Licensing information page available
+ at runtime from the About dialog for additional information.
+-->
+<template>
+<div>
+    <div v-if="canEdit">
+        <plot-options-edit />
+    </div>
+    <div v-if="!canEdit">
```
Comment (nikhilmandlik) [inline src/plugins/plot/vue/inspector/PlotOptions.vue:27]: `v-else` ? 😃

File: src/plugins/plot/vue/inspector/PlotOptionsBrowse.vue [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                     title="Manually override how the Y axis is labeled."
+                >Label</div>
+                <div class="grid-cell value">{{ config.yAxis.get('label') ? config.yAxis.get('label') : "Not defined" }}</div>
+            </li>
+            <li class="grid-row">
+                <div class="grid-cell label"
+                     title="Automatically scale the Y axis to keep all values in view."
+                >Autoscale</div>
+                <div class="grid-cell value">
+                    {{ config.yAxis.get('autoscale') ? "Enabled: " : "Disabled" }}
```
Comment (akhenry) [inline src/plugins/plot/vue/inspector/PlotOptionsBrowse.vue:45]: Would love it if we could avoid making function calls from templates.

File: src/plugins/plot/vue/inspector/PlotOptionsBrowse.vue [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    },
+    beforeDestroy() {
+        this.stopListening();
+    },
+    methods: {
+        getConfig() {
+            this.configId = this.openmct.objects.makeKeyString(this.domainObject.identifier);
+
+            return configStore.get(this.configId);
+
```
Comment (nikhilmandlik) [inline src/plugins/plot/vue/inspector/PlotOptionsBrowse.vue:143]: remove extra line

File: src/plugins/plot/vue/inspector/PlotOptionsEdit.vue [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    },
+    beforeDestroy() {
+        this.stopListening();
+    },
+    methods: {
+        getConfig() {
+            this.configId = this.openmct.objects.makeKeyString(this.domainObject.identifier);
+
+            return configStore.get(this.configId);
+
```
Comment (nikhilmandlik) [inline src/plugins/plot/vue/inspector/PlotOptionsEdit.vue:80]: same extra line 😄

File: src/plugins/plot/vue/inspector/PlotOptionsItem.vue [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    },
+    data() {
+        return {
+            expanded: false
+        };
+    },
+    computed: {
+        getSeriesClass() {
+            let cssClass = '';
+            let legacyObject = this.openmct.legacyObject(this.series.domainObject);
```
Comment (akhenry) [inline src/plugins/plot/vue/inspector/PlotOptionsItem.vue:98]: No legacy API please. Let's add a method to the Object API to determine if a given object path is to a link. I think the logic basically is "is the `location` property equal to the ID of the parent of this object"

File: src/plugins/plot/vue/inspector/PlotOptionsItem.vue [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+            let location = legacyObject.getCapability('location');
+            if (location && location.isLink()) {
+                cssClass = 'l-icon-link';
+            }
+
+            let type = legacyObject.getCapability('type');
```
Comment (akhenry) [inline src/plugins/plot/vue/inspector/PlotOptionsItem.vue:104]: No legacy API please, we have a Type API which should be able to give you the information you need here.

**Review by nikhilmandlik — DISMISSED** [review]
small changes else LGTM

File: src/plugins/timeline/TimelineViewLayout.vue (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/timeline/TimelineViewLayout.vue:68]: It's still being used to render the timeline axis in this view.

File: src/plugins/timeline/TimelineViewLayout.vue (hunk 2 above)
Comment (shefalijoshi) [reply src/plugins/timeline/TimelineViewLayout.vue:94]: TimelineObjectView is also being used in this view.

File: src/plugins/plot/vue/inspector/PlotOptions.vue (hunk 3 above)
Comment (shefalijoshi) [reply src/plugins/plot/vue/inspector/PlotOptions.vue]: Done.

File: src/plugins/plot/vue/inspector/PlotOptionsBrowse.vue (hunk 5 above)
Comment (shefalijoshi) [reply src/plugins/plot/vue/inspector/PlotOptionsBrowse.vue]: Done

File: src/plugins/plot/vue/inspector/PlotOptionsItem.vue (hunk 8 above)
Comment (shefalijoshi) [reply src/plugins/plot/vue/inspector/PlotOptionsItem.vue:104]: This has been taken care of in the legacy plots removal PR.

**Review by nikhilmandlik — APPROVED** [review]
LGTM

**Pull request merged**